common/rulesys.cpp

```diff
@@ -491,50 +480,36 @@ bool RuleManager::UpdateOrphanedRules(Database *db, bool quiet_update)
 	std::vector<std::string> rule_data;
 	std::vector<std::string> orphaned_rule_entries;
 
-	// load database rule names
-	std::string query("SELECT `rule_name` FROM `rule_values` GROUP BY `rule_name`");
-
-	auto results = db->QueryDatabase(query);
-	if (!results.Success()) {
+	const auto& l = RuleValuesRepository::GetGroupedRules(*db);
+	if (l.empty()) {
 		return false;
 	}
 
 	// build rule data entries
-	for (const auto &ri_iter : s_RuleInfo) {
-		if (strcasecmp(ri_iter.name, "Invalid Rule") == 0) {
+	for (const auto &r : s_RuleInfo) {
+		if (Strings::Contains(r.name, "Invalid Rule")) {
```

Comment:
This changes behavior as well

`strcasecmp` compares if strings are equal case insensitive whereas `Contains` is a partial

Reply:
Will use `Strings::Equals`.